lib/views/pages/home/content/ranking/ranking_widget.dart

```diff
@@ -1,10 +1,75 @@
+import "package:collection/collection.dart";
 import "package:flutter/material.dart";
+import "package:proxima/models/ui/ranking/ranking_details.dart";
+import "package:proxima/models/ui/ranking/ranking_element_details.dart";
+import "package:proxima/views/pages/home/content/ranking/ranking_card.dart";
+import "package:proxima/views/pages/home/content/ranking/ranking_list.dart";
 
+/// A class that displays the ranking widget.
 class RankingWidget extends StatelessWidget {
   const RankingWidget({super.key});
 
+  static const rankingListKey = Key("rankingList");
+  static const userRankingCardKey = Key("userRankingCard");
+
+  // Creating mock data for the ranking list
+  final List<Map<String, int>> userList = const [
+    {"user123": 9420},
+    {"johnDoe": 8370},
+    {"janeSmith": 6550},
+    {"coolGuy": 5280},
+    {"happyGal": 3000},
+    {"superman": 2310},
+    {"batman": 1530},
+    {"wonderWoman": 420},
+  ];
+
+  final RankingElementDetails userRankElementDetails =
+      const RankingElementDetails(
+    userDisplayName: "You",
+    userUserName: "U_You",
+    centauriPoints: 100,
+    userRank: null,
+  );
+
   @override
   Widget build(BuildContext context) {
-    return const Center(child: Text("Ranking"));
+    //Mock data creation (should be replaced with actual data)
+    final rankElementDetailsList = userList
+        .mapIndexed(
+          (index, user) => RankingElementDetails(
+            userDisplayName: user.keys.first,
+            userUserName: "u_${user.keys.first}",
+            centauriPoints: user.values.first,
+            userRank: index + 1,
```

Comment:
The user rank will already be provided with the correct exact rank. I think this should go outside the build method to stay consistent with where you defined your fake data.

Then you should just have a `final rankingDetails` (the one you have below) of type `RankingDetails` inside your build method and you only depend on this variable inside build.
That way when integrating with the viewmodel will come we will simply be able to change the expression of `rankingDetails` inside build using a provider.
Btw you can also add a TODO comment above this `rankingDetails` local variable.

Reply:
You are completely right ! It will make view model integration easier 💯 

Added in: [e482a05](https://github.com/ProximaEPFL/proxima/pull/322/commits/e482a05b8cc8aa25fad04f6804f98b94e3425782) and  [bc9ba87](https://github.com/ProximaEPFL/proxima/pull/322/commits/bc9ba8778fe2a198d85291ac050884e740ad18d0)

Thank you !